### PR TITLE
Add configurable "start page"

### LINKED ISSUE #1496: Add configurable "start page"
Allow a "start page" to be configured. This is the page that will be shown:

- on application start-up
- after the application is idle for a certain time, as configured in settings

Add a settings page at Settings > Display & Language > Start Page, with these options:

- Start page 
   Show this page on application start-up.
    - Auto (toggle)
    - Brief
    - Brief - side panel open
    - Overview
    - Levels - Tanks
    - Levels - Environment
    - Battery overview
    
- Timeout
  Revert to the start page when the application is inactive.
    - Never  <-- this is the default selection
    - 1 minute
    - 5 minutes
    - 10 minutes
    - 30 minutes
    - 60 minutes

### PR DESCRIPTION
**Pull request by blammit**
Allow a "start page" to be configured from Settings > Display & Language > Start Page. This is the page that will be shown:

- on application start-up 
- after the application is idle for a certain time

If the "Auto" option is selected, and a timeout is set, then the start page will be set automatically from the current application page after one minute, if it is one of the available options.

Fixes #1496

### CODE REVIEW

**Comment by blammit** [comment]
For convenience, in mock mode if you go to "Settings > Display & Language > Start Page" and then select an option, you can press "W" to go to that page immediately.

**Comment by blammit** [comment]
Not ready for review, need to fix the auto option here.

**Comment by blammit** [comment]
Updated, ready for review.

File: data/StartPageConfiguration.qml [hunk 1]
```diff
@@ -0,0 +1,200 @@
+/*
+** Copyright (C) 2023 Victron Energy B.V.
```
Comment (chriadam) [inline data/StartPageConfiguration.qml:2]: nitpick: 2024

**Comment by chriadam** [comment]
I built this for webassembly, and ran it locally on host (http://localhost:30000/venus-gui-v2.html?mqtt=ws://192.168.5.96:9001/) and it gets stuck after connecting to the MQTT broker, still on the splash page.

**Review by DanielMcInnes — APPROVED** [review]
lgtm

**Comment by DanielMcInnes** [comment]
> I built this for webassembly, and ran it locally on host (http://localhost:30000/venus-gui-v2.html?mqtt=ws://cerbo-ip:9001/) and it gets stuck after connecting to the MQTT broker, still on the splash page.

FYI I don't see this problem

File: data/StartPageConfiguration.qml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+							: VenusOS.StartPage_Type_Levels_Environment
+				}
+			}
+			break
+		}
+		return -1
+	}
+
+	// Whether the start page is enabled. 0 = disabled (i.e. auto-select), 1 = enabled (do not auto-select)
+	readonly property VeQuickItem _startPageMode: VeQuickItem {
```
[the remark this answers is not in the document]
Comment (blammit) [reply data/StartPageConfiguration.qml:187]: Yes let's move to doing that, the internal properties are just clutter sometimes

File: data/StartPageConfiguration.qml (hunk 1 above)
Comment (blammit) [reply data/StartPageConfiguration.qml:2]: Done

**Comment by blammit** [comment]
> I built this for webassembly, and ran it locally on host (http://localhost:30000/venus-gui-v2.html?mqtt=ws://cerbo-ip:9001/) and it gets stuck after connecting to the MQTT broker, still on the splash page.

Fixed, it happens if JSON.parse() succeeds but returns an empty value.

**Review by chriadam — APPROVED** [review]

**Pull request merged**